### ISSUE TITLE
Resolve Unnecessary Event (`Evt`) for Requests

### ISSUE THREAD

**Issue by HelloFillip** [cloudflare/kv-asset-handler#284]
Currently when constructing requests for assets with `getAssetFromKV` an `Evt` is needed. The `Event` holds the `Request` object and a `waitUntil` object (as a `Promise`). 

When using the package it was easier to omit the event and use `await` outside of the `getAssetFromKV` function so I omitted it and replaced the `Evt` parameter with a `Request` parameter, adding `await` in the cache persist:

```typescript
await cache.put(cacheKey, response.clone())
```

Noting that this does not result in exactly the same behavior but simplified the function calls. 

Looking back at this the `Evt` allows for additional control in the process but the lack of a simplified API could be resolved with either an alternative function that creates the `Evt` object implicitly, or exporting the `Evt` object and an `Evt` factory function to make the API more intuitive. Both approaches could be used, with the former suitable for cases when a resolvable `Promise` isn't needed, however the latter (an `Evt` factory function) would at least improve the API.

I've made these changes in a local fork already for my own project, so this issue is more about starting a discussion to put it into an acceptable PR, if possible.

**Comment by Cherry** [comment]
Using `waitUntil` for the `cache.put` is definitely preferred over `await`-ing it, since that holds up the response, whereas `waitUntil` does not. 

You can stub the `Evt` with something like:
```js
return getAssetFromKV(
  {
    request,
    waitUntil(promise) {
      return ctx.waitUntil(promise)
    },
  },
  ...
)
```

**Comment by HelloFillip** [comment]
Agreed. My internal solution has been to create an event (`Evt`) factory class:

```typescript
/**
 * Factory class for the Cloudflare Worker Event (`Evt`).
 */
class WorkerEvent {

    /** HTTP `Request` object. */
    request: Request;

    /** Closure when the cache is being persisted. */
    waitUntil: (promise: Promise<any>) => void;

    /**
     * Creates a new `WorkerEvent` with a HTTP request and closure for cache persistence management.
     * 
     * @param request HTTP `Request` object.
     * @param waitUntil Closure for when the cache persistence happens.
     */
    constructor(request: Request, waitUntil?: (promise: Promise<any>) => void) {
        this.request = request;

        if (waitUntil) {
            this.waitUntil = waitUntil;
        } else {
            this.waitUntil = (promise) => {
                promise;
            }
        }   
    }
}
```

Along with a simplified asset request function that negates the need for an explicit promise. 

```typescript
/**
 * Simplified interface to Cloudflare's `getAssetFromKV`.
 * 
 * @param input URL (`RequestInfo`) to use when constructing the HTTP `Request`.
 * @returns A promise resolvable to a `Response`.
 */
function getAsset(input: RequestInfo): Promise<Response> {
    const request = new Request(input);
    const event = new WorkerEvent(request);

    return getAssetFromKV(event);
}
```

The latter does not make sense to include in the library, but is useful for my own needs. The former is a convenient helper that enables me to create consistent `Evt` structures every time. 

This means the core code in the `getAssetFromKV` function remains the same, whilst an additional solution to creating the `Evt` is available in the API. The stub can be wrapped in a function rather than using a class with a constructor. Either way, the result would be similar; providing a helper for consistent `Evt` creation.
